### ISSUE TITLE
bundle exec anycable command is not doing anything

### ISSUE THREAD

**Issue by cesargomez89** [anycable/anycable#102]
**Ruby version: 2.7.1**

**`anycable` gem version: 1.0.0**

**`grpc` gem version: 1.28**

It works great in development environment,
but the command `RAILS_ENV=production bundle exec anycable`
is not doing anything in ubuntu server. It's just printing this:
```
$ RAILS_ENV=production bundle exec anycable
/shared/bundle/ruby/2.7.0/gems/anyway_config-2.0.5/lib/.rbnext/2.8/anyway/tracing.rb:176: warning: Passing the keyword argument as the last hash parameter is deprecated
/shared/bundle/ruby/2.7.0/gems/anyway_config-2.0.5/lib/.rbnext/2.8/anyway/tracing.rb:35: warning: The called method `record_value' is defined here
```
even using debug modes.

I don't know if that's the reason but the wss url is not working.
```
action_cable.js:241 WebSocket connection to 'wss://stage.xxxxxx.io:3334/cable' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
```

rpc systemd service
```
[Unit]

Description=anycable for staging
After=syslog.target network.target

[Service]

Type=simple
WorkingDirectory=/home/deploy/stage.xxxxxxx.io/current

ExecStart=/bin/bash -lc 'bundle exec anycable'
ExecStop=/bin/kill -TERM $MAINPID

User=deploy
Group=deploy
UMask=0002

MemoryHigh=2G
MemoryMax=3G
MemoryAccounting=true

RestartSec=1
Restart=on-failure

Environment=RAILS_ENV=production
Environment=ANYCABLE_PORT=3334
Environment=ANYCABLE_REDIS_URL='xxxxxxx'
Environment=ANYCABLE_HOST=stage.xxxxxxx.io
Environment=ANYCABLE_PATH=/cable
Environment=ANYCABLE_REDIS_CHANNEL=__anycable__

[Install]

WantedBy=multi-user.target
```

anycable GO service
```
[Unit]
Description=AnyCable Go WebSocket Server
After=network.target

[Service]
Type=simple
ExecStart=/usr/local/bin/anycable-go -headers=cookie,x-api-token,origin --ssl_cert=/etc/letsencrypt/live/stage.xxxxxxxx.io/fullchain.pem --ssl_key=/etc/letsencrypt/live/stage.xxxxxxxxxx.io/privkey.pem
ExecStop=/bin/kill -TERM $MAINPID

LimitNOFILE=16384 # increase open files limit (see OS Tuning guide)

Restart=on-failure

# Configure WebSocket server using env vars
Environment=RAILS_ENV=production
Environment=ANYCABLE_PORT=3334
Environment=ANYCABLE_REDIS_URL='xxxxxxxxxxxx'
Environment=ANYCABLE_HOST=stage.xxxxxxxxxx.io
Environment=ANYCABLE_PATH=/cable
Environment=ANYCABLE_REDIS_CHANNEL=__anycable__
# Environment=ANYCABLE_RPC_HOST=localhost:50051
# Environment=ANYCABLE_METRICS_HTTP=/metrics

[Install]
WantedBy=multi-user.target

```

services log
```
● anycable-go.service - AnyCable Go WebSocket Server
   Loaded: loaded (/etc/systemd/system/anycable-go.service; disabled; vendor preset: enabled)
   Active: active (running) since Wed 2020-07-01 21:52:40 UTC; 3min 9s ago
  Process: 7974 ExecStop=/bin/kill -TERM $MAINPID (code=exited, status=0/SUCCESS)
 Main PID: 7976 (anycable-go)
    Tasks: 8 (limit: 4915)
   CGroup: /system.slice/anycable-go.service
           └─7976 /usr/local/bin/anycable-go -headers=cookie,x-api-token,origin --ssl_cert=/etc/letsencrypt/live/stage.xxxxxxxxxx.io/fullchain.pem --ssl_key=/etc/letsencrypt/live/stage.xxxxxxx.io/privk

Jul 01 21:52:40 stage systemd[1]: Started AnyCable Go WebSocket Server.
Jul 01 21:52:40 stage anycable-go[7976]: I 2020-07-01T21:52:40.189Z context=main Starting AnyCable 1.0.0-33085f2 (pid: 7976, open file limit: 1024)
Jul 01 21:52:40 stage anycable-go[7976]: I 2020-07-01T21:52:40.190Z context=main Handle WebSocket connections at https://stage.xxxxxxxxxx.io:3334/cable
Jul 01 21:52:40 stage anycable-go[7976]: I 2020-07-01T21:52:40.190Z context=main Handle health connections at https://stage.xxxxxxxxx.io:3334/health
Jul 01 21:52:40 stage anycable-go[7976]: I 2020-07-01T21:52:40.191Z context=rpc RPC controller initialized: localhost:50051 (concurrency: 28, proto_versions: v0,v1)
Jul 01 21:52:40 stage anycable-go[7976]: I 2020-07-01T21:52:40.291Z context=pubsub Subscribed to Redis channel: __anycable__

● anycable.service - anycable for staging
   Loaded: loaded (/etc/systemd/system/anycable.service; disabled; vendor preset: enabled)
   Active: active (running) since Wed 2020-07-01 21:52:40 UTC; 3min 9s ago
  Process: 7975 ExecStop=/bin/kill -TERM $MAINPID (code=exited, status=0/SUCCESS)
 Main PID: 7995 (bundle)
    Tasks: 40 (limit: 4915)
   Memory: 114.6M (high: 2.0G max: 3.0G)
   CGroup: /system.slice/anycable.service
           └─7995 /var/www/stage.xxxxxxxxxx.io/shared/bundle/ruby/2.7.0/bin/anycable

Jul 01 21:52:40 stage systemd[1]: Stopped anycable for staging.
Jul 01 21:52:40 stage systemd[1]: Started anycable for staging.
Jul 01 21:52:41 stage bash[7995]: /var/www/stage.xxxxxxxxxxx.io/shared/bundle/ruby/2.7.0/gems/anyway_config-2.0.5/lib/.rbnext/2.8/anyway/tracing.rb:176: warning: Passing the keyword argument as the last ha
Jul 01 21:52:41 stage bash[7995]: /var/www/stage.xxxxxxxxxx.io/shared/bundle/ruby/2.7.0/gems/anyway_config-2.0.5/lib/.rbnext/2.8/anyway/tracing.rb:35: warning: The called method `record_value' is defined

```

environments/production.rb
```
  5   config.after_initialize do
  6     config.action_cable.url = ActionCable.server.config.url = 'wss://stage.xxxxxx.io:3334/cable'
  7     config.action_cable.allowed_request_origins = [
  8       "https://#{Current.domain}",
  9       "https://www.#{Current.domain}"
 10     ]
 11   end

```

Thanks in advance

**Comment by palkan** [comment]
> is not doing anything in ubuntu server. It's just printing this:

When running a Rails application, AnyCable uses Rails logger (so, `--debug` or `ANYCABLE_DEBUG` has almost no effect, it only enables gRPC logs). How is it configured?

>  WebSocket connection to 'wss://stage.xxxxxx.io:3334/cable' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED

Since we do not see any error logs for anycable-go, I believe that's a network issue; `ANYCABLE_HOST` is passed directly to Go's `ListenAndServe` function, which deep inside has an interesting comment regarding hostnames:

> The address can use a host name, but this is not recommended, because it will create a listener for at most one of the host's IP addresses.

As a quick fix, you can try to set `ANYCABLE_HOST=0.0.0.0`.

**Comment by cesargomez89** [comment]
Hey @palkan,
You were totally right, it was the `ANYCABLE_HOST`.
Thanks and regards from Colima, MX 🌮
